### PR TITLE
Add Physics Tests project

### PR DESCRIPTION
**Pull request by pouleyKetchoupp**
New project to provide a test framework for the 3D physics engines, based on part 1 of this proposal:
https://github.com/godotengine/godot-proposals/issues/667

I'm planning to add more tests in the coming weeks, but I'm providing this first version in order to start getting feedback.

### CODE REVIEW

File: 3d/physics_tests/Test.gd [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+func cancel_timer():
+	if timer_started:
+		timer.paused = true
+		timer.emit_signal("timeout")
+		timer.paused = false
+
+func is_timer_canceled() -> bool:
+	return timer.paused
+
+func on_timer_done():
```
Comment (aaronfranke) [inline 3d/physics_tests/Test.gd:30]: These scripts do not follow the GDScript style guide, there should be two lines between functions. Please read the whole style guide, there are several other issues that I can see: https://docs.godotengine.org/en/latest/getting_started/scripting/gdscript/gdscript_styleguide.html

File: 3d/physics_tests/main.tscn [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+[node name="Main" type="Control"]
+anchor_right = 1.0
+anchor_bottom = 1.0
+mouse_filter = 2
+script = ExtResource( 12 )
+__meta__ = {
+"_edit_use_anchors_": false
+}
+
+[node name="Tests_Menu" type="MenuButton" parent="."]
```
Comment (aaronfranke) [inline 3d/physics_tests/main.tscn:24]: Node names should be `PascalCase` with no underscore to match the convention of Godot's built-in names.

File: 3d/physics_tests/tests/functional/test_compound_shapes.tscn [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+[node name="MeshInstance" type="MeshInstance" parent="StaticBody_Plane"]
+transform = Transform( 50, 0, 0, 0, 1, 0, 0, 0, 50, 0, 0, 0 )
+mesh = SubResource( 3 )
+material/0 = null
+
+[node name="CollisionShape" type="CollisionShape" parent="StaticBody_Plane"]
+transform = Transform( 50, 0, 0, 0, 1, 0, 0, 0, 50, 0, 0, 0 )
+shape = SubResource( 4 )
+
+[node name="StaticBody_Head" type="StaticBody" parent="."]
```
Comment (aaronfranke) [inline 3d/physics_tests/tests/functional/test_compound_shapes.tscn:144]: The head and plane should be in their own scene, probably together.

File: 3d/physics_tests/Test.gd [hunk 4]
```diff
@@ -0,0 +1,31 @@
+extends Node
+class_name Test
+
+var timer : Timer
+var timer_started : bool = false
```
Comment (aaronfranke) [inline 3d/physics_tests/Test.gd:5]: @akien-mga says he doesn't want static typing used in the demos, so for now the policy I've been using is "no static typing unless it significantly improves readability".

File: 3d/physics_tests/README.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+They can be used for different purpose:
+- Functional tests to check for regressions and 
+compare the behavior between physics engines
+- Performance tests to evaluate and compare 
+performance between physics engines
+
+Language: GDScript
+
+Renderer: GLES 3
```
Comment (aaronfranke) [inline 3d/physics_tests/README.md:14]: Since rendering doesn't affect physics, the project should probably be converted to GLES 2.

**Comment by pouleyKetchoupp** [comment]
Thanks @aaronfranke for your review! I'll look into the GDScript style guide and fix the different things you pointed out.

**Comment by pouleyKetchoupp** [comment]
I've just pushed an update from the review:
- Switched to GLES2
- Godot head and plane and ground plane in a separate scene
- Style fixes in GDScript, scene nodes and project files

**Comment by pouleyKetchoupp** [comment]
Thanks! I'm interested to hear more details about the things you would like to add. I've got my own todo list for the next tests I want to implement, but I would gladly add more things if they can be useful.

**Comment by aaronfranke** [comment]
Mostly more complex scenarios, some are test cases for known existing physics bugs.

* Compound objects colliding with compound objects.

* Tests for 2D physics would be nice too.

* There's a bug where characters on a flat surface will slightly bounce up and down, it's visible in the Voxel demo and the TPS demo. The problem is most visible when you are colliding with more than one StaticBody (such as on a chunk border in the Voxel demo).

* There's a bug where adding colliders is extremely slow. In the Voxel demo, if you disable VSync, comment out line 186 in `chunk.gd`, and set gravity to 0 on line 8 in `player.gd` so that you don't fall forever, you can observe that the game runs vastly faster by not generating colliders.

For the last point, here is a minimal script which showcases the problem. If placed on a StaticBody by itself with nothing else in the scene (no cameras or anything), for me it starts at over 300 FPS, gradually slows down to about 30 FPS, then once it's done adding colliders I get over 9000 FPS.

```gdscript
extends StaticBody

var amount = 1000

func _process(delta):
	amount -= 1
	if amount > 0:
		var c = CollisionShape.new()
		c.shape = BoxShape.new()
		add_child(c)
```

**Pull request merged**